src/errors.rs

```diff
@@ -86,12 +85,22 @@ impl<'r> Responder<'r> for ApiError {
                 .status(http::Status::InternalServerError)
                 .ok()
         } else {
-            // otherwise, present the error in the requested data format
             let http_status = self.status();
             debug!("ApiError: {:?}", self);
-            Response::build_from(Output(self).respond_to(request)?)
+
+            match request.guard::<NegotiatedContentType>() {
```

Comment:
Why it's needed all of the sudden?

Reply:
Because we use `Output<T>` to serialize `ApiError`'s as well. W.o this, a 404 returned by this new route (i.e. if  you pass `Accept: text/plain; charset=utf-8`) will be partially replaced by `416 Not Acceptable`: the status will be correctly set to 404, but the body will come from https://github.com/xsnippet/xsnippet-api/blob/master/src/web/content.rs#L116-L119 (because `NegotiatedContentType` returns a failure).

I agree that it does look hacky :( But the cleanest approach I can think of is going all the way with supporting text serialization for everything (#140), or introducing a separate `MaybeTextOutput<T>` type, so that we can selectively return it in this route :(